elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/AsyncProperties.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.spring.config;
+
+import lombok.Data;
+
+/**
+ * Extra properties for setting up async query support.
+ */
+@Data
+public class AsyncProperties extends ControllerProperties {
+
+    /**
+     * Default thread pool size.
+     */
+    private int threadPoolSize = 6;
+
+    /**
+     * Default max query run time.
+     */
+    private int maxRunTime = 60;
```

Comment:
I think we'll want two times here - one for the timeout and one for cleanup (deletion).  Both should be configurable.

Reply:
We are calculating the cleanup time in elide-async based on provided maxRunTime. It's set to the maximum of 6 Hours or Twice of the maxRunTime provided by the user.  If we pass both params then we will have to do checks to make sure cleanup time is not less than the timeout.

Reply:
https://github.com/yahoo/elide/blob/c88d08a94c994f16ee51c6a51c39694411fb8c59/elide-async/src/main/java/com/yahoo/elide/async/service/AsyncExecutorService.java#L63